web/modules/custom/react_tools/tools/react-locator-map/src/components/searchForm.js

```diff
@@ -1,20 +1,44 @@
 import React, { Component } from 'react';
+import InputValidation from './inputValidation';
 
 class SearchForm extends Component {
 
   handleChange = (event) =>{
+    if(this.props.isInvalidFields){
+      this.props.setValidationFlagFn(false);
+    }
     this.props.setSearchLocationFn(event.target.value);
   }
 
   handleClick = (isLocationServices) =>{
-    if(!isLocationServices && !this.props.searchLocation) return;
+    if(!isLocationServices && !this.props.searchLocation) {
+      this.warningMessage = "Please fill out this field.";
+      this.props.setValidationFlagFn(true);
+      return;
+    }
+
+    if(this.props.isInvalidFields){
+      this.props.setValidationFlagFn(false);
+    }
     this.props.searchFn(isLocationServices);
   }
 
+  validationDisplay = (isInvalid, message) =>{
+    if(isInvalid){
+        return (
+           <InputValidation type="warning" message={message} />
+        )
+    }
+}
+
   render() {
+    let placeholderText = 'e.g. 62225, Fort Belvoir';
```

Comment:
Should this be const instead of let? since we are not changing the value of this variable within this scope

Reply:
it could be, it is only used in the enclosure so it really doesn't matter tbh